test/jdk/java/awt/image/MemoryLeakTest/MemoryLeakTest.java

```diff
@@ -22,154 +22,111 @@
-  this.name = name;
-  output = new TextArea(11, 50);
-  add("North", output);
-  output.append("Do the following steps on Solaris only.\n");
-  output.append("Maximize and minimize the Memory Leak Test window.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("Maximize and minimize the Memory Leak Test window again.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("If the two SZ values are the same, plus or minus one,\n");
-  output.append("then click Pass, else click Fail.");
-  Panel buttonPanel = new Panel();
-  passButton = new Button("Pass");
-  failButton = new Button("Fail");
-  passButton.addActionListener(this);
-  failButton.addActionListener(this);
-  buttonPanel.add(passButton);
-  buttonPanel.add(failButton);
-  add("South", buttonPanel);
-  pack();
-}
-
-public void start()
-{
-  show();
-}
+ * @bug 4078566 6658398
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test for a memory leak in Image.
+ * @run main MemoryLeakTest
+ */
 
-public void actionPerformed(ActionEvent event)
-{
-    if ( event.getSource() == passButton ) {
-      Globals.testPassed = true;
-      System.err.println(name + " Passed.");
+import java.awt.Color;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Image;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+
+//class Globals {
+//  static boolean testPassed = false;
+//    static Thread mainThread = null;
+//}
+
+public class MemoryLeakTest {
+    static Thread mainThread = null;
+    private static final String INSTRUCTIONS =
+        """
+         Do the following steps on Solaris only.
+         Maximize and minimize the Memory Leak Test window.
+         Execute the following after minimize.
+             ps -al | egrep -i 'java|PPID'
+         Examine the size of the process under SZ.
+         Maximize and minimize the Memory Leak Test window again.
+         Execute the following after minimize.
+             ps -al | egrep -i 'java|PPID'
+         Examine the size of the process under SZ.
+         If the two SZ values are the same, plus or minus one,
+         then click Pass, else click Fail.
+        """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+            .builder()
+            .title("MemoryLeakTest Instructions")
+            .instructions(INSTRUCTIONS)
+            .rows(15)
+            .columns(40)
+            .testUI(MemoryLeak::new)
+            .build()
+            .awaitAndCheck();
+        mainThread = Thread.currentThread();
+        if (mainThread != null) {
+            mainThread.interrupt();
+        }
```

Comment:
It is still unclear why we still need this. It seems to be a remnant of the old TestDialog machinery.

Reply:
I initially thought this may be needed, but looks like you're right. Removed, thanks for the help!